Sources/Core/AST/Stmt/ConditionalCompilationStmt.swift

```diff
@@ -0,0 +1,89 @@
+/// A conditional-compilation statement.
+public struct ConditionalCompilationStmt: Stmt {
+
+  public enum VersionComparison: Codable {
+
+    case greaterOrEqual
+
+    case less
+
+    func compare<T: Comparable>(_ lhs: T, _ rhs: T) -> Bool {
+      switch self {
+      case .greaterOrEqual: return !(lhs < rhs)
+      case .less: return lhs < rhs
+      }
+    }
+
+  }
+
+  public enum Condition: Codable, Equatable {
+
+    case `true`
+
+    case `false`
+
+    case os(Identifier)
+
+    case arch(Identifier)
+
+    case compiler(Identifier)
```

Comment:
```suggestion
    /// Holds iff the name of the compiler processing this file matches the payload.
    case compiler(Identifier)
```
Here I understand that we may need an arbitrary identifier.